test/gtest/common/main.cc

```diff
@@ -81,16 +81,23 @@ int main(int argc, char **argv) {
     UCS_TEST_MESSAGE << "Using random seed of " << ucs_gtest_random_seed;
     srand(ucs_gtest_random_seed);
     if (RUNNING_ON_VALGRIND) {
-        modify_config_for_valgrind("IB_RX_QUEUE_LEN", "512");
-        modify_config_for_valgrind("IB_RX_BUFS_GROW", "512");
-        modify_config_for_valgrind("MM_RX_BUFS_GROW", "128");
+        modify_config_for_valgrind("MM_RX_BUFS_GROW", "32");
+        modify_config_for_valgrind("MM_FIFO_SIZE", "32");
+        modify_config_for_valgrind("IB_ALLOC", "heap");
+        modify_config_for_valgrind("IB_RX_BUFS_GROW", "128");
         modify_config_for_valgrind("IB_TX_QUEUE_LEN", "128");
         modify_config_for_valgrind("IB_TX_BUFS_GROW", "64");
-        modify_config_for_valgrind("RC_TX_CQ_LEN", "256");
+        modify_config_for_valgrind("UD_RX_QUEUE_LEN", "256");
+        modify_config_for_valgrind("UD_RX_QUEUE_LEN_INIT", "16");
+        modify_config_for_valgrind("RC_TX_CQ_LEN", "128");
+        modify_config_for_valgrind("RC_RX_QUEUE_LEN", "128");
+        modify_config_for_valgrind("DC_TX_QUEUE_LEN", "16");
+        modify_config_for_valgrind("DC_MLX5_NUM_DCI", "3");
```

Comment:
maybe use common vars where it is possible?
Like `UCX_IB_TX_QUEUE_LEN`, `UCX_IB_RX_QUEUE_LEN`, etc

Reply:
it was needed to set different values for different transports
|   | RC  | DC  | UD  |  
|---|---|---|---|
| RX  |  IB_=128  |  DC_=16 |  IB_=128 |  
| TX |  IB_=28 |  RC_=128 |  UD_=256/16 |

Reply:
is there really a need to have different values under valgrind?

Reply:
- for DC, needed to reduce TX queue because there are 3 QPs per iface, but it wasn't possible on UD/RC because of TX moderation
- if set UD RX queue to 128, ai_md test hangs

Reply:
ok